src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ProcessLinkerXmlBase.cs

```diff
@@ -144,28 +177,44 @@ private void ProcessType(ModuleDesc assembly)
 
                 _reader.Read();
 
-                while (_reader.IsStartElement())
```

Comment:
Maybe let's move the part after this line into a new `protected virtual void ProcessType(TypeDesc type)` and override it for the ILLink.Descriptors-specific logic.